src/traces/image/attributes.js

```diff
@@ -0,0 +1,84 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var plotAttrs = require('../../plots/attributes');
+var hovertemplateAttrs = require('../../plots/template_attributes').hovertemplateAttrs;
+var extendFlat = require('../../lib/extend').extendFlat;
+
+module.exports = extendFlat({
+    z: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc',
+        description: [
+            'A 2-dimensional array in which each element is an array of 3 or 4 numbers representing a color.',
+        ].join(' ')
+    },
+    colormodel: {
+        valType: 'enumerated',
+        values: ['rgb', 'rgba', 'hsl', 'hsla'],
+        dflt: 'rgb',
+        role: 'info',
+        editType: 'plot',
+        description: 'Color model used to map the numerical color components described in `z` into colors.'
+    },
+    zmin: {
+        valType: 'data_array',
```

Comment:
We should make `zmin` and `zmax` have `valType: 'info_array'`:

https://github.com/plotly/plotly.js/blob/6e9bafb8285f7247407a503e7dc33e44ff637f31/src/lib/coerce.js#L254-L321

Reply:
Done in 73c0740f53fc82a70196d1993ba2824db2be85a3. Please let me know if it's :ok_hand:

Reply:
Moreover, @antoinerg can you a few test cases checking what happens when `zmin` and `zmax` are partially set, for example:

```js
var trace = {
  zmin: [0, 1],
  zmax: [0]
}
```

Reply:
In commit b998a17b249cfa93068e5ea719774d9357b45b1b, I removed the `dimensions` key and added tests to confirm that partially set `zmin` and `zmax` end up with a complete array!

Reply:
`dimensions` is still showing for `zmin`:

https://github.com/plotly/plotly.js/blob/b998a17b249cfa93068e5ea719774d9357b45b1b/src/traces/image/attributes.js#L43

Reply:
Oh my god, sorry about that :man_facepalming: 

Fixed in dec946902cdc69fbab792148730f77bbf9c0ade6

Reply:
No worries!! `zmin` and `zmax` are now looking :100: